rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+    ) -> ReplaceNodeOperatorPayload {
+        ReplaceNodeOperatorPayload {
+            node_ids: node_ids.to_vec(),
+            new_operator_id,
+            old_operator_id,
+        }
+    }
+
+    trait AssertHelpers {
+        #[track_caller]
+        fn assert_err_contains(self, expected: &str);
+
+        #[track_caller]
+        fn assert_ok(self);
+    }
+
+    impl<T> AssertHelpers for Result<T, String> {
+        #[track_caller]
+        fn assert_err_contains(self, expected: &str) {
+            match self {
+                Ok(_) => panic!("Expected error, but got Ok."),
+                Err(e) => assert!(
+                    e.contains(expected),
+                    "Expected error containing '{expected}', but got '{e}'"
+                ),
+            }
+        }
+
+        #[track_caller]
+        fn assert_ok(self) {
+            assert!(
+                self.is_ok(),
+                "Expected Ok, but got Err: {}",
+                self.err().unwrap()
+            )
+        }
+    }
+
+    fn mutation(key: Vec<u8>, value: Vec<u8>) -> RegistryMutation {
+        RegistryMutation {
+            mutation_type: Type::Upsert as i32,
+            key,
+            value,
+        }
+    }
+
+    fn operator_mutation(
+        operator: PrincipalId,
+        provider: PrincipalId,
+        node_allowance: u64,
+        dc_id: &str,
+    ) -> RegistryMutation {
+        let operator_record = NodeOperatorRecord {
+            node_operator_principal_id: operator.as_slice().to_vec(),
+            node_allowance,
+            node_provider_principal_id: provider.as_slice().to_vec(),
+            dc_id: dc_id.to_string(),
+            ..Default::default()
+        };
+
+        mutation(
+            make_node_operator_record_key(operator).as_bytes().to_vec(),
+            operator_record.encode_to_vec(),
+        )
+    }
+
+    thread_local! {
+        /// Needed for invariant checks of each node
+        /// because each node has to have unique xnet
+        /// and http endpoints
+        static NODE_NUM: RefCell<u8> = const { RefCell::new(0) };
+    }
+
+    fn node_mutation(node_id: NodeId, operator: PrincipalId) -> RegistryMutation {
+        let current_node_num = NODE_NUM.with_borrow_mut(|current| {
+            *current += 1;
+            *current
+        });
+        let node_record = NodeRecord {
+            node_operator_id: operator.as_slice().to_vec(),
+            xnet: Some(ConnectionEndpoint {
+                ip_addr: format!("192.{current_node_num}.0.1"),
+                port: 8080,
+            }),
+            http: Some(ConnectionEndpoint {
+                ip_addr: format!("192.{current_node_num}.0.2"),
+                port: 8080,
+            }),
+            ..Default::default()
+        };
+
+        mutation(
+            make_node_record_key(node_id).as_bytes().to_vec(),
+            node_record.encode_to_vec(),
+        )
+    }
+
+    fn dc_mutation(dc_id: &str) -> RegistryMutation {
```

Comment:
again, this name is not specific enough.  I have to read it to tell what it does, which means that in the context of a test, it's actually making it harder for me as the reader to understand what the test setup is doing.

Reply:
I hope that renaming it helps, curious to see what you think after the renames 😄